### PR TITLE
fix(commons): cover VLM raw image proto validation

### LINKED ISSUE #895: Add proto-adapter tests for VLM raw_rgb/raw_rgba validation
Follow-up from #814 by @shubhamsinnh -- thanks again for that PR!

## What
#814 added exact-size checks, overflow guards, and null-alloc handling to the raw_rgb/raw_rgba branches of `rac_vlm_image_from_proto` (core/src/foundation/rac_proto_adapters.cpp), but none of it runs in CI: both `VLMImage` instances in `core/tests/test_advanced_modality_proto_abi.cpp` use `set_file_path` (lines 535 and 730), and no test builds a raw_rgb or raw_rgba payload.

## Why it matters
A future refactor of this adapter could silently reintroduce the heap over-read/overflow #814 fixed, and the linux-asan `ctest` leg (pr-build.yml:172) would not catch it because nothing exercises this path.

## Suggested approach
Add proto-adapter test cases for raw_rgb/raw_rgba covering: a length mismatch, width or height <= 0, dimensions that overflow `size_t` on a 32-bit build, and an exact-size RGBA buffer checked byte-for-byte after alpha is dropped.

## Done when
- [ ] `test_advanced_modality_proto_abi.cpp` (or a new test file) covers all four cases above.
- [ ] `ctest --preset linux-asan` exercises the new raw_rgb/raw_rgba paths.

Not blocking #814. @shubhamsinnh, you know this code well now -- you're welcome to take this one if you're interested.

_Opened with help from Claude Code and Codex._

_Reviewed with help from Claude Code and Codex._

### PR DESCRIPTION
**Pull request by shubhamsinnh**
## Summary

- cover exact-size RGB copying and reject undersized and oversized RGB payloads
- reject non-positive raw-image dimensions and conditionally exercise 32-bit size overflow paths
- verify RGBA-to-RGB alpha removal byte-for-byte and reject oversized RGBA payloads

## Testing

- git diff --check
- local GCC 16.1/Ninja Debug build of test_advanced_modality_proto_abi
- local compile of the updated test object after review changes
- Linux ASan configure, build, and complete CTest preset in PR CI
- local Windows CTest was blocked before test entry by a MinGW loader error (0xc0000139)

Fixes #895

### CODE REVIEW

File: core/tests/test_advanced_modality_proto_abi.cpp [hunk 1]
```diff
@@ -112,6 +114,101 @@ bool serialize(const google::protobuf::MessageLite& message, std::vector<uint8_t
...
+          "raw RGB conversion preserves dimensions and size");
+    CHECK(out.pixel_data != nullptr &&
+              std::memcmp(out.pixel_data, rgb_bytes.data(), rgb_bytes.size()) == 0,
+          "raw RGB conversion copies every byte");
+    rac_free(const_cast<uint8_t*>(out.pixel_data));
+
+    runanywhere::v1::VLMImage short_rgb;
+    short_rgb.set_width(2);
+    short_rgb.set_height(1);
+    short_rgb.set_raw_rgb(std::string(5, '\x01'));
```
[the remark this answers is not in the document]
Comment (shubhamsinnh) [reply core/tests/test_advanced_modality_proto_abi.cpp:138]: Covered in cc70f5ccc: both RGB and RGBA now have one-byte-oversized payload cases that assert rejection and an empty output buffer. The updated test object compiles successfully.
